src/geo/map.js

```diff
@@ -230,25 +234,37 @@ cdb.geo.Layers = Backbone.Collection.extend({
    * the index should be recalculated
    */
   _assignIndexes: function(model, col, options) {
-    var layerTypeWeight = {
-      'torque': 100
-    };
-    function layerWeight(layer) {
-      var t = layer.get('type');
-      return layerTypeWeight[t] || 0;
-    }
-    var from = 0;//this.size() - 1;
-    if(options && options.at !== undefined) {
-      from = options.at;
-    }
-    if(from === 0) {
+    var TILED_LAYER_TYPE = 'Tiled';
+    var CARTODB_LAYER_TYPE = 'CartoDB';
+    var TORQUE_LAYER_TYPE = 'torque';
+
+    if (this.size() > 0) {
+
+      // Assign an order of 0 to the first layer
       this.models[0].set({ order: 0 });
```

Comment:
Is this.models a Backbone.Collection? If so, we should make use of their API, this.models.at(0) (super picky thing and it is not necessary buddy).